src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferenceCache.cs

```diff
@@ -32,29 +32,44 @@ public static async ValueTask<FindReferenceCache> GetCacheAsync(Document documen
 
         static async Task<FindReferenceCache> ComputeCacheAsync(Document document, CancellationToken cancellationToken)
         {
+            var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
the cache is an object we make for a document *once* we've decided it's a doc of interest and we're going ot be searching it.  that means it already registered a 'hit' during the pass when we asked to find docments that could potnetially contain the symbols we're searching for (say because it got a hit in the bloomfilter of all the names in the doc).

Once we've decided we're looking at the doc, it makes sense to get all the info we need up front, so that hte search in the doc doesn't ever ahve to go async.  So we do this here.  Note: teh *txt* is the most normal thing we need in a normal find refs.  Say we're lookign for `Foo`  and we have a hit in a doc.  We actually use the text to find the positions in the text to try to find tokens to actually bind.   So really, there's no benefit deferring these things since we will practically *always* need them.